sympy/integrals/integrals.py

```diff
@@ -1404,6 +1406,79 @@ def integrate(*args, **kwargs):
     else:
         return integral
 
+def cauchy_principal_value(func, sym, lower_limit = -oo, upper_limit = oo):
+    """
+    Compute the Cauchy Principle Value of the definite integral of a real function on the entire real axis.
+    In mathematics, the Cauchy principal value, is a method for assigning values to certain improper
+    integrals which would otherwise be undefined.
+    Cauchy Principle Value is  defined as:
```

Comment:
Principle -> Principal

Reply:
I'll correct this @jksuom . Thanks for raising this point.